core/EE_System.core.php

```diff
@@ -394,6 +394,7 @@ public function load_espresso_addons()
                 array('activate', 'activate-selected'),
                 true
             )
+            && ($this->request->isWordPressApi() || $this->request->isApi())
```

Comment:
this is great but would be slightly better if this conditional block was the first one in the entire set, since it is the easiest and therefore fastest to evaluate, ie:

```php
        if (($this->request->isWordPressApi() || $this->request->isApi())
            && $this->request->getRequestParam('activate') !== 'true'
            && ! function_exists('json_basic_auth_handler')
            && ! function_exists('json_basic_auth_error')
            && ! in_array(
                $this->request->getRequestParam('action'),
                array('activate', 'activate-selected'),
                true
            )
        ) {
            include_once EE_THIRD_PARTY . 'wp-api-basic-auth' . DS . 'basic-auth.php';
        }
```

Reply:
y kk!